core/src/main/java/org/mule/runtime/core/internal/policy/CommonSourcePolicy.java

```diff
@@ -86,29 +85,23 @@ public void process(CoreEvent sourceEvent,
     });
   }
 
-  public MessageSourceResponseParametersProcessor getResponseParamsProcessor(CoreEvent event) {
-    return ((InternalEvent) event).getInternalParameter(POLICY_SOURCE_PARAMETERS_PROCESSOR);
-  }
-
   public void finishFlowProcessing(CoreEvent event, Either<SourcePolicyFailureResult, SourcePolicySuccessResult> result) {
     if (!((BaseEventContext) event.getContext()).isComplete()) {
       ((BaseEventContext) event.getContext()).success(event);
     }
 
-    recoverCallback(event).complete(result);
+    from(event).getProcessCallback().complete(result);
```

Comment:
instead of doing `getProcessCallback().complete(result);`, why not have the `complete` method in `SourcePolicyContext`?